client/src/globals/UserContext.js

```diff
@@ -8,6 +8,22 @@ export const UserContext = createContext({
 export const UserContextProvider = (props) => {
     const [userId, setUserId] = useState("");
     const value = { userId, setUserId };
+    useEffect(() => {
```

Comment:
I guess we can leave this here for now? For me, it also makes more sense to move it to private route.

Reply:
Right, but I just thought about it again, and leave this in userContext, might help with identify if user already logged in or not (ex. when user navigate to login page manually when the user already logged in, frontend should navigate user to home page). However, we can still do this process in other ways tho.